sasgrid-and-viya/azuredeploy.json

```diff
@@ -0,0 +1,5833 @@
+                        "createOption": "FromImage",
+                        "caching": "ReadWrite",
+                        "managedDisk": {
+                            "storageAccountType": "Premium_LRS"
+                        },
+                        "diskSizeGB": 64
+                    },
+                    "dataDisks": [
+                        {
+                            "name": "[concat(variables('ansible_vm_name'),'-playbook')]",
+                            "diskSizeGB": 50,
+                            "lun": 0,
+                            "createOption": "Empty"
+                        }
+                    ]
+                },
+                "osProfile": {
+                    "computerName": "[variables('ansible_vm_name')]",
+                    "adminUsername": "[parameters('primaryUserName')]",
+                    "linuxConfiguration": {
+                        "disablePasswordAuthentication": true,
+                        "ssh": {
+                            "publicKeys": [
+                                {
+                                    "path": "[concat('/home/', parameters('primaryUserName'), '/.ssh/authorized_keys')]",
+                                    "keyData": "[parameters('SSHPublicKey')]"
+                                }
+                            ]
+                        }
+                    }
+                },
+                "networkProfile": {
+                    "networkInterfaces": [
+                        {
+                            "id": "[resourceId('Microsoft.Network/networkInterfaces', variables('ansible_nw_interface'))]"
+                        }
+                    ]
+                },
+                "diagnosticsProfile": {
+                    "bootDiagnostics": {
+                        "enabled": true,
+                        "storageUri": "[reference(variables('diagnostic_storagegroup_name')).primaryEndpoints.blob]"
+                    }
+                },
+                "proximityPlacementGroup": {
+                    "id": "[resourceId('Microsoft.Compute/proximityPlacementGroups',variables('ppg_name'))]"
+                }
+            }
+        },
+        {
+            "type": "Microsoft.Resources/deployments",
+            "apiVersion": "2019-10-01",
+            "name": "AnsibleRoleAssignment",
+            "dependsOn": [
+                "[resourceId('Microsoft.Compute/virtualMachines', variables('ansible_vm_name'))]"
+            ],
+            "properties": {
+                "mode": "Incremental",
+                "templateLink": {
+                    "uri": "[variables('role_assignment_template')]"
+                },
+                "parameters": {
+                    "roleAssignmentName": {
+                        "value": "[guid(resourceId('Microsoft.Compute/virtualMachines', variables('ansible_vm_name')), variables('reader_role'), variables('rbacPrincipalID'))]"
```

Comment:
rbacPrincipalId should be the value of the principalId below

Reply:
we have used rbacprincipalId, just as a variable for reference. Removed it to avoid confusion.

Reply:
Still not correct - the principalId that is used in the name must be the same as the principalId used in the assignment.  I.e. the managedIdentity value you have below